### ISSUE TITLE
Conflict between local and remote file every time a file in external storage is edited

### ISSUE THREAD

**Issue by wonx** [nextcloud/desktop#3913]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

### Expected behaviour
When I edit a file that is located on an external storage, let's say a text file, I enter some text inside and save. The file get synced and uploaded to the server. I should be able to do that multiple times, and the most recent file should be the good one.

### Actual behaviour
When I save the file for the second time, nextcloud tries to sync it but it will complain that there is a conflict, and I have to go to the Desktop Client and choose between the local and the remote version of the file. That happens for every edition and for many type of files (I have tried several libreoffice documents).

![Captura de 2021-10-20 17-18-46](https://user-images.githubusercontent.com/2736639/138121944-8791a987-333d-4492-87c4-c3835dbc50a5.png)

Note: it seems to work correctly in native nextcloud folders. This bug only applies to folders stored in external storage (sftp in my case).

### Steps to reproduce
1. Open or create a text file and save it on a folder that is stored on an external storage
2. Write some text inside, and save.
3. Wait until it syncs
4. Write some more text, and save again.
5. Wait until it syncs again. A conflict message will appear on the desktop client.

### Client configuration
Client version: 3.3.5 (appimage)

Operating system: Ubuntu 20.04 LTS

OS language: Catalan

Qt version used by client package (Linux only, see also Settings dialog): (couldn't find it)

Client package (From Nextcloud or distro) (Linux only): Nextcloud-3.3.5-x86_64.AppImage

Installation path of client: /home//Applications/

### Server configuration

Nextcloud version: 22.1.0 and mysql 8

Storage backend (external storage): sftp

**Comment by wonx** [comment]
This problem is still existing.

**Comment by wonx** [comment]
I can confirm the problem is also reproducible in the Windows 10 client. 

Also, I made sure that the problem was not due to locked files (by checking the presence of locked files in the database).

**Comment by claell** [comment]
Is this a duplicate of #2467?

**Comment by wonx** [comment]
It could be, but I don't know if internally they are caused by different processes.

**Comment by wonx** [comment]
Could it be also related to https://github.com/nextcloud/text/issues/1896 ? It seems quite similar.

**Comment by wonx** [comment]
I can confirm this problem still appears after upgrading the client to version 23.0.0

**Comment by ruedigerkupper** [comment]
Have we confirmed whether this is a duplicate of #2467 or a different bug?

**Comment by ruedigerkupper** [comment]
I see the same 412 HTTP errors (precondition for PUT failed) in the logs of my sync client.

**Comment by TheNickSkater** [comment]
I have pretty much the same issue but with Office files (word, excel, pp etc).
I can most of the time save once, but the second time it runs into a conflict.
Client version: 3.4.2
NextCloud: 23.0.1

I can't say since when this happens, because I did not use it in such a scenario since like half a year.

**Comment by ruedigerkupper** [comment]
I am almost sure this is a duplicate of #2467. I think we should mark this bug as duplicate.

**Comment by wonx** [comment]
Yes, it's mostly when you save a file for the second time. The only workaround I found is to pause the sync before starting working with files, and turn it back on once I have finished. Otherwise I'd get conflicts.

**Comment by ruedigerkupper** [comment]
That's exactly the symptoms we see in #2467.
@wonx Would you mind closing this as a duplicate of #2467?

**Comment by ruedigerkupper** [comment]
Focussing this in one single report might give us more attention by developers.

**Comment by wonx** [comment]
Duplicate of #2467
